### ISSUE TITLE
deserialize uses dubious test for reader vs bytes

### ISSUE THREAD

**Issue by jackpunt** [thesayyn/protoc-gen-ts#44]
There are cases where (bytes instanceof Uint8Array) fails.
More robust to reverse the test, and look for definite BinaryReader.

from:
````
        const reader = bytes instanceof Uint8Array ? new pb_1.BinaryReader(bytes) : bytes, message = new XXX();
````
to:
````
        const reader = bytes instanceof pb_1.BinaryReader ? bytes : new pb_1.BinaryReader(bytes), message = new XXX();
````

My auto-patching tricked me into thinking this was fixed back in -rc3

**Comment by thesayyn** [comment]
Yeah... this is something that I forgot to get done when fixing other things.  I'll quickly send a patch to get this done.

**Comment by thesayyn** [comment]
A fix for this has already landed on master via https://github.com/thesayyn/protoc-gen-ts/commit/31a79cacf31bd48b7610c19942ce7e1c38bec84b
